sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Filtering/Filter.cs

```diff
@@ -458,28 +473,28 @@ private Expression ProduceComparatorExpressionForSingleFieldCondition(Expression
                                 case Predicate.LessThan:
                                     // (int)fieldValue < (int)enumValue
                                     // (int?)fieldValue < (int?)enumValue
-                                    Type underlyingType = isFieldTypeNullable ? typeof(Nullable<>).MakeGenericType(enumUnderlyingType) : enumUnderlyingType;
```

Comment:
The `MakeGenericType` here isn't trimmable.

But since we know all the possible types here (defined in the case statements at lines 423-432), we can replace the `MakeGenericType` with a switch statement (line 450-463).